### PR TITLE
TM-5218, Create Printable Panel Agenda

### PR DESCRIPTION
**Pull request by M00staff**
[Ticket](https://metaphase.atlassian.net/browse/TM-5218)

image

image

image

### CODE REVIEW

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    'Employee Challenge': [],
+  };
+
+  const categorizeAgendas = () => {
+    agendas.forEach(a => {
+      agendasCategorized[meetingCategoryMap[a?.pmi_mic_code]].push(a);
+    });
+    return agendasCategorized;
+  };
+
```
Comment (M00staff) [inline src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:48]: ^^ a lot of the above is copy pasted logic from `PanelMeetingAgendas.jsx`

Comment (M00staff) [inline src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:48]: ^^ a lot of the above is copy pasted logic from `PanelMeetingAgendas.jsx`

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    'Position Challenge': [],
+    'Employee Challenge': [],
+  };
+
+  const categorizeAgendas = () => {
+    agendas.forEach(a => {
+      agendasCategorized[meetingCategoryMap[a?.pmi_mic_code]].push(a);
+    });
+    return agendasCategorized;
+  };
```
Comment (M00staff) [inline src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:47]: ^^ a lot of the above is copy pasted logic from `PanelMeetingAgendas.jsx`

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                          </tr>
+                        </thead>
+                        { printableAgendaTable(agenda) }
+                      </table>
+
+                      <div className="pma-footer-wrapper">
+                        <div className="pma-cdo-remarks-wrapper">
+                          <div className="item"><span className="label">CDO: </span> {cdo}</div>
+                          <div className="item"><span className="label">Remarks: </span> {remarks}</div>
+                          <div className="item"><span className="label">Combined TOD: </span> {combinedTod}</div>
```
Comment (M00staff) [inline src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:164]: combined TOD wasn't in the design, so i added it - I THINK i got everything else that's on the Panel Meeting Agenda page though

when there was a discrepancy between the design's wording & what's currently on the page - i went with how it's currently rendered on the page 🤷 - i think it'd be off if it said one thing on the page and another on the print version

File: src/sass/_panelMeetingAgendas.scss [hunk 4]
```diff
@@ -229,3 +229,150 @@
...
+  .profile-menu {
+    display: none;
+  }
+  .search-bar-hidden {
+    display: none;
+  }
+  .tm-footer {
+    display: none;
+  }
+}
```
Comment (M00staff) [inline src/sass/_panelMeetingAgendas.scss:245]: ^^ these are global, so i don't know if they should be in this file 🤔

this was the simplest way to get the print working cleanly - and i dont think if a user was printing anything else on the site, they would mind if these were not rendered 🤷

File: src/sass/_panelMeetingAgendas.scss [hunk 5]
```diff
@@ -229,3 +229,150 @@
...
+    font-weight: bold;
+    padding: 10px 0 10px 0;
+    margin: 2px 0 2px 0px;
+    border-top: solid 5px $primary-gray-dark;
+    border-bottom: solid 5px $primary-gray-dark;
+  }
+  .pma-table-wrapper {
+    box-shadow: $tm-gray 6px 6px 20px;
+    margin-bottom: 15px;
+    break-inside: avoid;
```
Comment (M00staff) [inline src/sass/_panelMeetingAgendas.scss:283]: SUPER neat CSS property

https://developer.mozilla.org/en-US/docs/Web/CSS/break-inside

File: src/sass/_panelMeetingAgendas.scss [hunk 6]
```diff
@@ -229,3 +229,150 @@
...
+  }
+
+  .pma-print-history-status {
+    font-family: 'Roboto';
+    display: flex;
+    -webkit-print-color-adjust: exact;   // Chrome, Safari 6 – 15.3, Edge
+    print-color-adjust: exact;           // Firefox 97+, Safari 15.4+
```
Comment (M00staff) [inline src/sass/_panelMeetingAgendas.scss:290]: Forces these elements to to actually render
image

Like this when printing
image

without having to click "Background Graphics" in Chrome
image

Prolly not necessary - but then I'd have to change from the white font when printing 🤷

I dont have a printer so no idea how it'd look either way

File: src/sass/_panelMeetingAgendas.scss (hunk 4 above)
Comment (M00staff) [reply src/sass/_panelMeetingAgendas.scss:245]: also didn't know `@media print` was an MQ ... Neat!

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+                        <div className={`poly-slash agenda-tag--${agendaStatus}`} />
+                      </div>
+
+                      <div className="pma-print-user-info">
+                        <div className="item"><span className="label">{name}</span></div>
+                        <div className="item"><span className="label">Bureau: </span> {userBureau}</div>
+                        <div className="item"><span className="label">Grade: </span> {userGrade}</div>
+                        <div className="item"><span className="label">Skill: </span> {userSkill}</div>
+                        <div className="item">
+                          <span className="label">Languages:</span>
```
Comment (jlam-mpc) [inline src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:128]: Needs space after colon to match padding as the other labels

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx (hunk 3 above)
Comment (jlam-mpc) [reply src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:164]: Good call

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+                            }
+                          </span>
+                        </div>
+                      </div>
+
+                      <table className="pma-print-table">
+                        <thead>
+                          <tr>
+                            <th>Action</th>
+                            <th>Org</th>
```
Comment (jlam-mpc) [inline src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:144]: Does this need to be Location/Org?

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+                            <th>Position Number</th>
+                            <th>Position Title</th>
+                            <th>Grade</th>
```
Comment (jlam-mpc) [inline src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:147]: Does this need to be PP/Grade?

**Review by jlam-mpc — COMMENTED** [review]
Everything looks pretty good I just had a few cosmetic related comments and also a few questions about the requirements.

I know you mentioned there's a few discrepancies with the design vs what's currently rendered on the feature but I would double check what the final consensus is for it because Charles' recent UI PR handles updating some of these fields too so you may have to make sure they are consistent.

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx (hunk 9 above)
Comment (M00staff) [reply src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:147]: that's funny - this one and the location/ORG one were the two fields i was specifically talking about that were different on the Panel Agenda page vs the design. 

went ahead and updated them both - doesn't hurt to add the slash value, so long as we don't rename it completely imo 🤷

File: src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+              <td>{leg.pay_plan}</td>
+            </tr>
+          </tbody>
+        );
+      })
+    );
+  };
+
+  return (
+    <div className="pma-print-view">
```
[the remark this answers is not in the document]
Comment (M00staff) [reply src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:82]: add 30px of padding

[the remark this answers is not in the document]
Comment (M00staff) [reply src/Components/Panel/PanelMeetingAgendas/PrintPanelMeetingAgendas.jsx:82]: good call by the way - cleans it up quite a bit!

**Comment by M00staff** [comment]
@jlam-mpc made all the updates!

**Review by jlam-mpc — APPROVED** [review]

**Pull request merged**